src/Interpreters/RewriteFunctionToSubcolumnVisitor.cpp

```diff
@@ -0,0 +1,52 @@
+#include <Interpreters/RewriteFunctionToSubcolumnVisitor.h>
+#include <DataTypes/NestedUtils.h>
+#include <Parsers/ASTIdentifier.h>
+
+namespace DB
+{
+
+namespace
+{
+
+ASTPtr transformToSubcolumn(const ASTIdentifier & identifier, const String & subcolumn_name)
+{
+    return std::make_shared<ASTIdentifier>(Nested::concatenateName(identifier.name(), subcolumn_name));
+}
+
+ASTPtr transformIsNotNullToSubcolumn(const ASTIdentifier & identifier, const String & subcolumn_name)
+{
+    auto ast = transformToSubcolumn(identifier, subcolumn_name);
+    ast = makeASTFunction("NOT", ast);
+    return ast;
+}
+
+const std::unordered_map<String, std::pair<String, decltype(&transformToSubcolumn)>> function_to_subcolumn =
+{
+    {"length",    {"size0", transformToSubcolumn}},
+    {"isNull",    {"null", transformToSubcolumn}},
```

Comment:
So if column is not `Nullable` that it will fail with an error, right? I guess it not great, and fallback to regular column should be done

Reply:
Yes, I just forgot, that `isNull` is also applicable to non nullable types. Anyway, it's better to always check types of rewritten columns. I've fixed it in fresh commit.